apollo-common/src/main/java/com/ctrip/framework/apollo/common/utils/UrlUtils.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.common.utils;
+
+import com.google.common.base.Strings;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Abner
+ * @since  8/31/22
+ */
+public final class UrlUtils {
+    private static final String ILLEGAL_KEY_REGEX = "[/\\\\]+|^\\.";
+
+    private UrlUtils() {}
+
+    public static boolean hasIllegalChar(String key) {
+        if(Strings.isNullOrEmpty(key)){
+            return false;
+        }
+        final Pattern pattern = Pattern.compile(ILLEGAL_KEY_REGEX, Pattern.MULTILINE);
+        final Matcher matcher = pattern.matcher(key);
+        return matcher.find();
+    }
+
+    public static String encode(String key){
+        return new String(Base64.getEncoder().encode(key.getBytes(StandardCharsets.UTF_8)));
+    }
+
+    public static String decode(String key){
+        return new String(Base64.getDecoder().decode(key.getBytes(StandardCharsets.UTF_8)));
+    }
```

Comment:
These are simply base64 encode and decode logic. I don't think it's necessary to have a common util for them.

Reply:
but if we change another algorithm, it should modify in many position, it's easily forgotten in somewhere.